src/widget/widget.js

```diff
@@ -0,0 +1,75 @@
+/**
+ * @license Copyright (c) 2003-2016, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+import Plugin from '../../core/plugin.js';
+import WidgetEngine from './widgetengine.js';
+import MouseObserver from '../../engine/view/observer/mouseobserver.js';
+import ModelRange from '../../engine/model/range.js';
+import { isWidget } from './utils.js';
+
+/**
+ * The widget plugin.
+ * Adds default {@link engine.view.Document#mousedown mousedown} handling on widget elements.
+ *
+ * @memberOf image.widget
+ * @extends core.Plugin.
+ */
+export default class Widget extends Plugin {
+	/**
+	 * @inheritDoc
+	 */
+	static get requires() {
+		return [ WidgetEngine ];
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	init() {
+		const viewDocument = this.editor.editing.view;
+
+		// If mouse down is pressed on widget - create selection over whole widget.
+		viewDocument.addObserver( MouseObserver );
+		this.listenTo( viewDocument, 'mousedown', ( ...args ) => this._onMousedown( ...args ) );
+	}
+
+	/**
+	 * Handles {@link engine.view.Document#mousedown mousedown} events on widget elements.
+	 *
+	 * @param {utils.EventInfo} eventInfo
+	 * @param {envine.view.observer.DomEventData} domEventData
+	 * @private
+	 */
+	_onMousedown( eventInfo, domEventData ) {
+		let widgetElement = domEventData.target;
+		const editor = this.editor;
+		const viewDocument = editor.editing.view;
+
+		// If target is not a widget element - check if one of the ancestors is.
+		if ( !isWidget( widgetElement ) ) {
```

Comment:
We might add `includeNode` option to `findAncestor` similar as in: https://github.com/ckeditor/ckeditor5-engine/blob/master/src/view/node.js#L129.